tests/masonry/builders/fixtures/with_optional_without_extras/pyproject.toml

```diff
@@ -0,0 +1,19 @@
+[project]
+name = "my-packager"
+description = "Something"
+version = "0.1"
+classifiers = [
+    "Topic :: Software Development :: Build Tools",
+    "Topic :: Software Development :: Libraries :: Python Modules"
+]
+dynamic = ["dependencies"]
+
+[tool.poetry.dependencies]
+requests = { version = "^0.28.1", optional = true }
+httpx = { version = "^0.28.1", optional = true }
+grpcio = { version = "^0.2.0" }
```

Comment:
This test case seems to show that it is ok not to bother marking grpc - or any other extra - as optional.  Is that the intention?

I would expect that a mismatch (in either direction) between declared extras and "optional" dependencies should result in an error, not in quietly taking a guess as to what was meant?

Reply:
Yes, I have noticed that, too. And I thought that at least `poetry check` should give a warning. On the other side, we deprecated `poetry.extras` anyway.

Reply:
Yes, this is place where the PEP621 syntax improves on poetry's - by making such mistakes impossible.

Reply:
I just added that case for coverage of status quo to be honest and to verify that the change did not accidentally regress the case.

But yes, PEP 621makes it impossible for this.